services/cd-service/pkg/cmd/server.go

```diff
@@ -344,6 +345,11 @@ func RunServer() {
 					logger.FromContext(ctx).Fatal("Could not pull repository to perform custom migrations", zap.Error(err))
 				}
 				logger.FromContext(ctx).Sugar().Warnf("running custom migrations, because KUBERPULT_DB_WRITE_ESL_TABLE_ONLY=false")
+
+				// we overwrite InsertApp in order to also update the overview:
+				dbHandler.InsertAppFun = func(ctx context.Context, transaction *sql.Tx, appName string, previousEslVersion db.EslVersion, stateChange db.AppStateChange, metaData db.DBAppMetaData) error {
```

Comment:
This creates a pretty weird control flow as the dbhandler is also used in the repo service. The first instance that does the migration will update the overviews and all later instances won't.

Reply:
I agree with the weird control flow. But, I assumed because of the recursive dependencies in go there's no quick workaround for that.

Reply:
`InsertAppFun` is only called when creating an app or deleting an app.
It will never be called in the migration, the migrations just call `DBInsertApplication` directly without updating the overview.

I agree the flow is not great, but I'm also not sure how to make this simpler.
And we have the same pattern a few lines below, so I think it's okay for now.